app/assets/src/components/ui/controls/SearchBoxList.jsx

```diff
@@ -0,0 +1,103 @@
+import React from "react";
+import PropTypes from "prop-types";
+import CheckmarkIcon from "ui/icons/CheckmarkIcon";
+import Input from "ui/controls/Input";
+import cs from "./search_box_list.scss";
+import cx from "classnames";
+
+class SearchBoxList extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      filter: "",
+      selected: new Set(this.props.selected)
+    };
+
+    this.sortedOptions = this.sortOptions();
+  }
+
+  handleOptionClick = optionValue => {
+    let selected = new Set(this.state.selected);
+    if (selected.has(optionValue)) {
+      selected.delete(optionValue);
+    } else {
+      selected.add(optionValue);
+    }
+    this.setState({ selected }, () => {
+      return this.props.onChange && this.props.onChange(this.state.selected);
+    });
+  };
+
+  handleFilterChange = filter => this.setState({ filter });
+
+  sortOptions() {
+    // TODO(tcarvalho): review data structures to simplify this function
+    let selectedOptions = {};
+    let unselectedOptions = [];
+    this.props.options.forEach(option => {
+      if (this.state.selected.has(option.value)) {
+        selectedOptions[option.value] = option;
+      } else {
+        unselectedOptions.push(option);
+      }
+    });
+    return Array.from(this.state.selected)
+      .map(optionValue => selectedOptions[optionValue])
+      .concat(unselectedOptions);
+  }
+
+  render() {
+    return (
+      <div className={cs.searchBoxList}>
+        {this.props.title && <div className={cs.title}>{this.props.title}</div>}
+        <div>
+          <Input
+            fluid
+            className={cs.searchBox}
+            icon="search"
+            placeholder="Search"
+            onChange={this.handleFilterChange}
+          />
+        </div>
+        <div className={cs.listBox}>
+          {this.sortedOptions.map(
+            option =>
+              option.label.toLowerCase().includes(this.state.filter) && (
```

Comment:
Was a little surprised to see the filtering happening here, but it does work.

If we were worried about performance issues, we might want to store the filteredOptions in a state variable instead (and compute them in the `handleFilterChange function).